src/mscorlib/src/System/Char.cs

```diff
@@ -929,10 +927,10 @@ public static String ConvertFromUtf32(int utf32)
             }
             // This is a sumplementary character.  Convert it to a surrogate pair in UTF-16.
             utf32 -= UNICODE_PLANE01_START;
-            char[] surrogate = new char[2];
-            surrogate[0] = (char)((utf32 / 0x400) + (int)CharUnicodeInfo.HIGH_SURROGATE_START);
-            surrogate[1] = (char)((utf32 % 0x400) + (int)CharUnicodeInfo.LOW_SURROGATE_START);
-            return (new String(surrogate));
+            char* surrogate = stackalloc char[2];
+            surrogate[0] = (char)((utf32 / 0x400) + HIGH_SURROGATE_START);
+            surrogate[1] = (char)((utf32 % 0x400) + LOW_SURROGATE_START);
+            return new string(surrogate, 0, 2);
```

Comment:
@hughbe Maybe, but 1) that's more of an implementation detail (it wouldn't work if you took the code and copy/pasted it outside of mscorlib), and 2) it requires pinning/unpinning the string while we write characters to it, so the benefit from that would be questionable. For now, I'm sticking to `stackalloc`.